plugins/weather/src/test/java/de/unistuttgart/iaas/amyassist/amy/plugin/weather/WeatherRestTest.java

```diff
@@ -44,54 +50,101 @@
 
 	@Reference
 	private TestFramework testFramework;
+	
+	private WeatherDarkSkyAPI logic;
+	private WeatherReportDay day;
+	private WeatherReportWeek week;
+	private JsonObject obj;
 
-	@Reference
-	private Server server;
-
-	@Reference
-	private WeatherResource weatherLogic;
-
-	private HttpServer httpServer;
 	private WebTarget target;
 
 	@BeforeEach
 	public void setUp() {
 		this.testFramework.setRESTResource(WeatherResource.class);
+		this.logic = this.testFramework.mockService(WeatherDarkSkyAPI.class);
 
 		Client c = ClientBuilder.newClient();
 		this.target = c.target(Server.BASE_URI);
+		
+		this.createDay();
+		this.createWeek();
 	}
-
+	
+	private void createDay() {
+		FIODataPoint p = Mockito.mock(FIODataPoint.class);
+		Mockito.when(p.summary()).thenReturn("Clear throughout the day");
+		Mockito.when(p.precipProbability()).thenReturn((double) 0);
+		Mockito.when(p.precipType()).thenReturn("no data");
+		Mockito.when(p.temperatureMin()).thenReturn((double) 15);
+		Mockito.when(p.temperatureMax()).thenReturn((double) 25);
+		Mockito.when(p.sunriseTime()).thenReturn("05:00:00");
+		Mockito.when(p.sunsetTime()).thenReturn("21:00:00");
+		Mockito.when(p.timestamp()).thenReturn((long) 12345);
+		
+		this.day = new WeatherReportDay("This is the weather report for today.", p);
+		
+		this.obj = new JsonObject();
+		this.obj.add("summary", this.day.getSummary());
+		this.obj.add("precip", this.day.isPrecip());
+		this.obj.add("precipProbability", this.day.getPrecipProbability());
+		this.obj.add("precipType", this.day.getPrecipType());
+		this.obj.add("temperatureMin", this.day.getTemperatureMin());
+		this.obj.add("temperatureMax", this.day.getTemperatureMax());
+		this.obj.add("sunriseTime", this.day.getSunriseTime());
+		this.obj.add("sunsetTime", this.day.getSunsetTime());
+		this.obj.add("weekday", this.day.getWeekday());
+		this.obj.add("timestamp", this.day.getTimestamp());
+	}
+	
+	private void createWeek() {
+		this.week = Mockito.mock(WeatherReportWeek.class);
+		this.week.summary = "Clear throughout the day";
+	}
+	
 	/**
 	 * Test method for
 	 * {@link de.unistuttgart.iaas.amyassist.amy.plugin.weather.WeatherResource#getWeatherToday()}.
 	 */
 	@Test
-	void testGetWeatherReportToday() {
-		// TODO
+	void testGetWeatherToday() {
+		Mockito.when(this.logic.getReportToday()).thenReturn(this.day);
+		
+		Response response = this.target.path("weather").path("today").request().get();
+
+		assertEquals(200, response.getStatus());
+		assertEquals(this.obj.toString(), response.readEntity(String.class));
 	}
 
 	/**
 	 * Test method for
 	 * {@link de.unistuttgart.iaas.amyassist.amy.plugin.weather.WeatherResource#getWeatherTomorrow()}.
 	 */
 	@Test
-	void testGetWeatherReportTomorrow() {
-		// TODO
+	void testGetWeatherTomorrow() {
+		Mockito.when(this.logic.getReportTomorrow()).thenReturn(this.day);
+		
+		Response response = this.target.path("weather").path("tomorrow").request().get();
+		
+		assertEquals(200, response.getStatus());
```

Comment:
use `assertThat(actuale, matcher)` like `assertThat(response.getStatus(), is(200))`